Dockerfile

```diff
@@ -83,25 +83,18 @@ v${OSTICKET_VERSION}/osTicket-v${OSTICKET_VERSION}.zip; \
         wget -q -O /var/www/html/include/i18n/${lang}.phar \
             https://s3.amazonaws.com/downloads.osticket.com/lang/1.14.x/${lang}.phar; \
     done
-ENV OSTICKET_PLUGINS_VERSION=cc4bf22f17cc0bebd1e1958786a271f02d7f2645 \
-    OSTICKET_PLUGINS_SHA256SUM=c34f9bee929eb256d62aee57b458d89e6a8dc258948a5850f9f314aa44fd20d8
 RUN set -ex; \
     \
-    wget -q -O osTicket-plugins.tar.gz https://github.com/devinsolutions/osTicket-plugins/archive/\
-${OSTICKET_PLUGINS_VERSION}.tar.gz; \
-    echo "${OSTICKET_PLUGINS_SHA256SUM}  osTicket-plugins.tar.gz" | sha256sum -c; \
-    tar -xzf osTicket-plugins.tar.gz --one-top-level --strip-components 1; \
-    rm osTicket-plugins.tar.gz; \
-    \
-    cd osTicket-plugins; \
-    php make.php hydrate; \
-    find * -maxdepth 0 -type d ! -path doc ! -path lib -exec mv '{}' \
-        /var/www/html/include/plugins +; \
-    cd ..; \
-    \
-    rm -r osTicket-plugins /root/.composer
-ENV OSTICKET_SLACK_VERSION=cd98e54fcadf1a5dd8e78b0a0380561c7ef29b02 \
-    OSTICKET_SLACK_SHA256SUM=9cdead701fd1be91a64451dfaca98148b997dc4e5a0ff1a61965bffeebd65540
+    for plugin in audit auth-2fa auth-ldap auth-passthru auth-password-policy storage-fs; do \
+        wget -q -O /var/www/html/include/plugins/${plugin}.phar \
+            https://s3.amazonaws.com/downloads.osticket.com/plugin/${plugin}.phar; \
+    done; \
+    for plugin in auth-oauth2 storage-s3; do \
+        wget -q -O /var/www/html/include/plugins/${plugin}.phar \
+            https://s3.amazonaws.com/downloads.osticket.com/plugin/1.17.x/${plugin}.phar; \
+    done
```

Comment:
We should try to authenticate these packages. Aren't they signed by the creators? I think I remember seeing some kind of indication of unauthenticated plugins in the admin panel.

Reply:
You are probably confusing this indication with the one for language packs. osTicket implements functions for validating signatures of PHARs here: https://github.com/osTicket/osTicket/blob/1.17.x/include/class.plugin.php#L466.

However, I couldn't find any usage of this function related to verifying plugins. It appears to be used only to verify language pack signatures.

To be sure, I downloaded a language pack and two plugins, and I ran the following code in the osTicket directory:

```php
<?php
define('INCLUDE_DIR', 'include/');
require('include/class.plugin.php');

foreach (array('es_ES.phar', 'audit.phar', 'auth-oauth2.phar') as $phar) {
	print("Checking " . $phar . "\n");
	$result = PluginManager::isVerified($phar);
	print("Result: " . $result . "\n\n");
}
?>
```

The result was:

```
Checking es_ES.phar
PHP Warning:  Undefined array key 1 in /home/adam/osTicket/include/class.plugin.php on line 489
Result: 1

Checking audit.phar
PHP Warning:  dns_get_record(): An unexpected server failure occurred. in /home/adam/osTicket/include/class.plugin.php on line 486
Result: 3

Checking auth-oauth2.phar
PHP Warning:  dns_get_record(): An unexpected server failure occurred. in /home/adam/osTicket/include/class.plugin.php on line 486
Result: 3
```

(1 is `VERIFIED`, 3 is `VERIFY_FAILED`).

The reason is that they are getting the signature of a checksum from a DNS TXT record at `<checksum>.updates.osticket.com`. The checksum for language packs is computed using SHA-1, but for plugins, it's SHA-256. The hexadecimal representation of a SHA-256 checksum is 65 characters long, but the maximum length of a DNS label is 63 characters. Therefore the signature can't be verified.

Reply:
`auth-oauth2.phar` has `FE075FA9783389744A613DD730FCFAB69A4AF5791238EF90649569A7DB0D55D6` SHA-256 hash signature and `FE075FA9783389744A613DD730FCFAB69A4AF5791238EF90649569A7DB0D55D.updates.osticket.com` (the hash is trimmed to 63 characters) returns no TXT records. Looks like [the algorithm](https://github.com/osTicket/osTicket/blob/03f8813348067f1700aee3ebbb8d7bd685a62694/include/class.plugin.php#L595) supports SHA-1 only anyway. We should:

1. Raise an issue with the maintainers and perhaps suggest embedding the signatures into Phars instead. Maybe we should add a TODO into `Dockerfile` referencing the issue.
2. At least create an issue in this repo. for verifying integrity of the language packs, as they contain executable code as well.

Reply:
> At least create an issue in this repo. for verifying integrity of the language packs, as they contain executable code as well.

I've implemented verification of integrity of all `.phar`s (both language packs, and plugins) we download in https://github.com/devinsolutions/docker-osticket/pull/38/commits/2f9e998ce8d58aeed470274e12beaf4982fe079b.

Also, there were some changes - it would appear that `auth-oauth2.phar` is now using `SHA-1` checksum, but `Plugin::isVerified` (`PluginManager` changed to `Plugin`) still won't succeed.

Reply:
1. Opened: https://github.com/osTicket/osTicket/issues/6376
2. Done in https://github.com/devinsolutions/docker-osticket/pull/38/commits/bf88654b08cc5617dd60c3177894c746245c8796. However, language packs for `ar_EG`, `ar_SA` and `az` don't pass the authenticity check (they also show as "Unverified" in the UI, I've opened https://github.com/osTicket/osTicket/issues/6377), so I removed them in that commit. Should I add them back and check only their integrity?
3. The TXT record does not exist:

```
$ php -r '$p = new Phar("auth-oauth2.phar"); print_r($p->getSignature());'         
Array
(
    [hash] => DFA8E6A04101DB5672AD17E8DE76E427C3BF0604
    [hash_type] => SHA-1
)
$ drill txt DFA8E6A04101DB5672AD17E8DE76E427C3BF0604.updates.osticket.com
;; ->>HEADER<<- opcode: QUERY, rcode: NXDOMAIN, id: 43761
;; flags: qr rd ra ; QUERY: 1, ANSWER: 0, AUTHORITY: 1, ADDITIONAL: 0 

[... SNIP ...]
```

Reply:
I'd leave out the language packs that can't be authenticated. For plugins, I'd add a TODO referencing osTicket/osTicket#6376.

Reply:
I've added the reference to the Dockerfile.